x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -58,21 +58,32 @@ environment variables to run inside of the container.
 
 The following actions are possible and grouped based on the actions.
 
-* Preparing Kibana for Fleet
-  This prepares the Fleet plugin that exists inside of Kibana. This must either be enabled here or done externally
-  before Fleet Server will actually successfully start.
+* Elastic Agent Fleet Enrollment
+  This enrolls the Elastic Agent into a Fleet Server. It is also possible to have this create a new enrollment token
+  for this specific Elastic Agent.
 
-  KIBANA_FLEET_SETUP - set to 1 enables this setup
-  KIBANA_FLEET_HOST - kibana host to enable Fleet on [$KIBANA_HOST]
-  KIBANA_FLEET_USERNAME - kibana username to enable Fleet [$KIBANA_USERNAME]
-  KIBANA_FLEET_PASSWORD - kibana password to enable Fleet [$KIBANA_PASSWORD]
-  KIBANA_FLEET_CA - path to certificate authority to use with communicate with Kibana [$KIBANA_CA]
+  FLEET_ENROLL - set to 1 for enrollment into fleet-server. If not set, Elastic Agent is run in standalone mode.
+  FLEET_URL - URL of the Fleet Server to enroll into
+  FLEET_ENROLLMENT_TOKEN - token to use for enrollment.
+  FLEET_CA - path to certificate authority to use with communicate with Fleet Server [$KIBANA_CA]
+  FLEET_INSECURE - communicate with Fleet with either insecure HTTP or un-verified HTTPS
+
+  The following vars are need in the scenario that Elastic Agent should automatically fetch its own token.
+
+  KIBANA_FLEET_HOST - kibana host to enable create enrollment token on [$KIBANA_HOST]
+  KIBANA_FLEET_USERNAME - kibana username to create enrollment token [$KIBANA_USERNAME]
+  KIBANA_FLEET_PASSWORD - kibana password to create enrollment token [$KIBANA_PASSWORD]
```

Comment:
I assumed that agent won't communicate directly with Kibana, so we won't need these vars?

Reply:
These are needed for the case where the Elastic Agent setups Fleet and fleet-server. In this case, Kibana access is needed. But this is more for convience in testing and setup in Cloud.